vm_operator/src/ratings.py

```diff
@@ -0,0 +1,88 @@
+"""Control App Center Ratings on a host system. Provides a Ratings class."""
+
+import logging
+
+from charms.operator_libs_linux.v1 import snap
+
+logger = logging.getLogger(__name__)
+
+
+class Ratings:
+    """Class representing Ratings on a host system."""
+
+    def install(self):
+        """Install the Ratings snap package."""
+        try:
+            self._snap.ensure(snap.SnapState.Latest, channel="edge")
+            snap.hold_refresh()
+        except snap.SnapError as e:
+            logger.error("could not install ratings. Reason: %s", e.message)
+            logger.debug(e, exc_info=True)
+            raise e
+
+    def refresh(self):
+        """Refresh the Ratings snap if there is a new revision."""
+        # The operation here is exactly the same, so just call the install method
+        self.install()
+
+    def start(self):
+        """Start and enable Ratings using the snap service."""
+        self._snap.start(enable=True)
+
+    def stop(self):
+        """Stop Ratings using the snap service."""
+        self._snap.stop(disable=True)
+
+    def remove(self):
+        """Remove the Ratings snap."""
+        self._snap.ensure(snap.SnapState.Absent)
+
+    def configure(
+        self,
+        jwt_secret=None,
+        log_level=None,
+        postgres_uri=None,
+        migration_postgres_uri=None,
+        env=None,
+        restart=False,
+    ):
+        """Configure Ratings on the host system. Restart Ratings by default."""
+        if jwt_secret:
+            self._snap.set({"app-jwt-secret": jwt_secret})
+            restart = True
+
+        if log_level:
+            self._snap.set({"app-log-level": log_level})
+            restart = True
+
+        if postgres_uri:
+            self._snap.set({"app-postgres-uri": postgres_uri})
+            restart = True
+
+        if migration_postgres_uri:
+            self._snap.set({"app-migration-postgres-uri": migration_postgres_uri})
+            restart = True
+
+        if env:
+            self._snap.set({"app-env": env})
+            restart = True
+
+        # Restart the snap service
+        if restart:
+            self._snap.restart()
```

Comment:
Maybe just drop all the `restart=True` above and leave `self._snap.restart()` here - no need to differentiate if all the config options need to restart!

```suggestion
        self._snap.restart()
```